### PR TITLE
Install openshift RPMs based on cluster version

### PR DESCRIPTION
**Pull request by mtnbikenc**
Retrieves current cluster version and installs openshift RPMs with
matching version.

https://jira.coreos.com/browse/CORS-1087

### CODE REVIEW

**Comment by mtnbikenc** [comment]
```
Failing tests:

[sig-storage] PersistentVolumes-local  Pods sharing a single local PV [Serial] all pods should be running [Suite:openshift/conformance/serial] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
Investigating tests.
/hold

**Comment by mtnbikenc** [comment]
```
Failure talking to yum: failure: repodata/repomd.xml from criotested: [Errno 256] No more mirrors to try.
    https://use-mirror1.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
    https://use-mirror2.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
    https://mirror.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
    https://euw-mirror1.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: [Errno 12] Timeout on https://euw-mirror1.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: (28, 'Connection timed out after 30002 milliseconds')
    https://gce-mirror1.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: [Errno 12] Timeout on https://gce-mirror1.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: (28, 'Connection timed out after 30001 milliseconds')
    https://gce-mirror2.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: [Errno 12] Timeout on https://gce-mirror2.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: (28, 'Connection timed out after 30001 milliseconds')
    https://euw-mirror1.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: [Errno 12] Timeout on https://euw-mirror1.ops.rhcloud.com/enterprise/rhel/cri-o-tested/4.2-rhel-7/x86_64/os/repodata/repomd.xml: (28, 'Connection timed out after 30001 milliseconds')
```

File: images/installer/root/usr/local/bin/entrypoint-provider [hunk 1]
```diff
@@ -24,6 +24,9 @@ if ! whoami &>/dev/null; then
   echo "${USER:-default}:x:$(id -u):$(id -g):Default User:$HOME:/sbin/nologin" >> /etc/passwd
 fi
 
+# Override cluster version check when running in CI
+mkdir -p "${WORK}/inventory/dynamic/${TYPE}/group_vars/all"
```
Comment (vrutkovs) [inline images/installer/root/usr/local/bin/entrypoint-provider]: Dupe of line #31?

File: roles/openshift_node/tasks/install.yml [hunk 2]
```diff
@@ -6,12 +6,42 @@
...
-  poll: 30
+- name: Get cluster version
+  command: >
+    oc get clusterversion
+    --config={{ openshift_node_kubeconfig_path }}
+    --output=jsonpath='{.items[0].status.desired.version}'
+  delegate_to: localhost
+  register: oc_get
+  until:
+  - oc_get.stdout != ''
```
Comment (vrutkovs) [inline roles/openshift_node/tasks/install.yml]:
```suggestion
  - oc_get.stdout is defined
  - oc_get.stdout != ''
```

Just like in #11595

**Comment by mtnbikenc** [comment]
Waiting for updated 4.2 crio RPM.
/hold

**Comment by DanyC97** [comment]
@mtnbikenc 👋  if you don't mind asking:  i was under the impression in 3.x that the `openshift-hyperkube` rpm should be replaced by the container image and as such i thought that moving forward in 4.x the rpm should not be a _thing_ ?

or i guess i made a wrong assumption? ;)

**Comment by mtnbikenc** [comment]
@DanyC97 The kubelet is still required and is installed by the openshift-hyperkube RPM.

**Comment by DanyC97** [comment]
> @DanyC97 The kubelet is still required and is installed by the openshift-hyperkube RPM.

i understand @mtnbikenc , thank you for taking the time to respond.

**Comment by mtnbikenc** [comment]
New crio 1.14.5.
/retest
/hold cancel

**Comment by mtnbikenc** [comment]
Resolved merge conflict.

**Comment by vrutkovs** [comment]
/hold

Putting it on hold to have a new template applied

**Comment by mtnbikenc** [comment]
```
Failing tests:

[Conformance][Area:Networking][Feature:Router] The HAProxy router converges when multiple routers are writing status [Suite:openshift/conformance/parallel/minimal]
[Feature:OpenShiftAuthorization][Serial] authorization  TestAuthorizationResourceAccessReview should succeed [Suite:openshift/conformance/serial]
[sig-autoscaling] [HPA] Horizontal pod autoscaling (scale resource: CPU) [sig-autoscaling] ReplicationController light Should scale from 2 pods to 1 pod [Suite:openshift/conformance/parallel] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
namespace ci-op-sbqdymw4 deleted
/hold cancel
/retest

**Comment by vrutkovs** [comment]
/hold
needs rebase

**Comment by mtnbikenc** [comment]
Rebased.
/hold cancel

**Pull request merged**